### DISCUSSION TITLE
how can I set the icon size as percentage in TS?

### DISCUSSION THREAD

**Discussion by GhostOrder28** [segmentio/evergreen#1450] [Q&A] [answered]
Previously with just javascript it wasn't an issue at all setting the size like this: 

`<ArrowUpIcon size={'100%'} />`

But now I'm migrating my app to typescript and the compiler says about the size attribute that `'string' is not assignable to type 'number'` which makes sense but then how can I set the size of icons as percentages?. 

I tried removing completely the size attribute and adding the `ub-h_100prcnt and ub-w_100prcnt` classes which are the ones evergreen adds to svgs when they have their sizes set  like `size={'100%'}` but, although the classes were added they were nullified(I couldn't find a better word) by the default `ub-h_16px and ub-w_16px` classes.

![Screenshot_20220419_153308](https://user-images.githubusercontent.com/73717218/164090839-eb6759e5-0a2b-4af8-90e9-1a01e015aab5.png)

**Comment by brandongregoryscott** [comment, marked as answer]
Hey @GhostOrder28! I think we typically expect icons to be sized in units (i.e. a `number`)
https://github.com/segmentio/evergreen/blob/10e8e2399029a22526b55e3f532a0c0b81952696/index.d.ts#L3183-L3187

If you have a specific use-case for a percentage or string-based value (and it was working as you expected), it should still work by casting it:

`<ArrowUpIcon size={('100%' as any) as number} />`

As for setting the class names directly, I would not recommend that - `ui-box` and `glamor` both manage helper class creation/application and aren't really meant to be used directly in the way you described.

**Reply by GhostOrder28** [reply]
Hey @brandongregoryscott thank you for the help, in this case I just want the size of the icons to depend on the size of their parent container.

**Reply by brandongregoryscott** [reply]
Gotcha. There are some calculations in the component logic that look like they expect a number, but if it's working, go for it!

https://github.com/segmentio/evergreen/blob/10e8e2399029a22526b55e3f532a0c0b81952696/src/icons/src/Icon.js#L23-L25
